File: exemplo-notificacao-nodejs/src/aplicacao.ts
import { Notificacao } from "./notificacao";

export class Aplicacao {
	private pilhaNotificacoes = [];

	getNotificacoes() {
		return this.pilhaNotificacoes;
	}

	async configurarNovaNotificacao(notificacao: Notificacao): Promise<boolean> {
		try {
			this.pilhaNotificacoes.push(notificacao);

			return true;
		} catch(error) {
			console.error(error)
			return false;
		}
	}

	async enviarNotificacoes(): Promise<void> {
		if (this.pilhaNotificacoes.length > 0) {
			for (const notificacao of this.pilhaNotificacoes) {
				await (
					(): Promise<void> =>
						new Promise(resolve => setTimeout(resolve, 500))
				)();

				notificacao.enviarNotificacao();
			}
		}
	}
}

File: exemplo-notificacao-python/aplicacao.py
from notificacao import Notificacao
from time import sleep as delay


class Aplicacao:
    def __init__(self) -> None:
        self.pilha_notificacoes: list[Notificacao] = []

    def get_notificacoes(self) -> list[Notificacao]:
        return self.pilha_notificacoes

    def configurar_nova_notificacao(self, notificacao: Notificacao) -> bool:
        try:
            self.pilha_notificacoes.append(notificacao)
            return True
        except Exception:
            return False

    def enviar_notificacoes(self) -> None:
        for notificacao in self.pilha_notificacoes:
            notificacao.enviar_notificacao()
            delay(0.5)
